cmd/ingester/app/consumer/consumer_test.go

```diff
@@ -86,9 +86,17 @@ func newConsumer(
 	factory metrics.Factory,
 	topic string,
 	processor processor.SpanProcessor,
-	consumer consumer.Consumer) *Consumer {
+	consumer consumer.Consumer,
+	maxReadsPerSecond float64) *Consumer {
 
 	logger, _ := zap.NewDevelopment()
+
+	var rateLimiter *time.Ticker
```

Comment:
Could you add more test cases? 
For e.g., 
If the rate limiter is configured to 0, does it allow any requests?
If the rate limiter is configured to 10qps, does it allow 10 requests?
If the rate limiter is configured to 1qps, and it receives 10 requests, does it allow the 10 requests in 10 seconds?

Reply:
Ya definitely. I agree this deserves direct testing.